src/server/json_family.cc

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/json_family.h"
+
+extern "C" {
+#include "redis/object.h"
+}
+
+#include "base/logging.h"
+#include "server/command_registry.h"
+#include "server/error.h"
+#include "server/tiered_storage.h"
+#include "server/transaction.h"
+
+#include <jsoncons/json.hpp>
+#include <jsoncons_ext/jsonpath/jsonpath.hpp>
+#include <absl/strings/str_join.h>
+
+namespace dfly {
+
+using namespace std;
+using namespace jsoncons;
+
+using CI = CommandId;
+
+namespace {
+
+string GetString(EngineShard* shard, const PrimeValue& pv) {
+  string res;
+  if (pv.IsExternal()) {
+    auto* tiered = shard->tiered_storage();
+    auto [offset, size] = pv.GetExternalPtr();
+    res.resize(size);
+
+    error_code ec = tiered->Read(offset, size, res.data());
+    CHECK(!ec) << "TBD: " << ec;
+  } else {
+    pv.GetString(&res);
+  }
+
+  return res;
+}
+
+OpResult<string> OpGet(const OpArgs& op_args, string_view key, vector<pair<string_view, jsonpath::jsonpath_expression<json>>> expressions) {
```

Comment:
please change vector argument to const ref. no need to copy all this by value.
I would also create an alias `using JsonExpression = jsonpath::jsonpath_expression<json>` around lines 24-25 to make the code a bit more succinct.

Reply:
ah, I see now you moved the vector. Why did you move and not use const-ref? Does json library require for the expression to be mutable?

Reply:
Indeed, expression objects should be mutable. I can try to cast expressions with const_cast to override the constness, but the code wouldn't be reliable...

Reply:
oh no, that's not needed :)